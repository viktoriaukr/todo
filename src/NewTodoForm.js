import React, { useState } from "react";
import { v4 as uuid } from "uuid";

const NewTodoForm = ({ addTodo }) => {
  const [formData, setFormData] = useState({ task: "" });
  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo({ ...formData, id: uuid() });
    setFormData({ task: "" });
  };
  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="task">Todo</label>
      <input
        id="task"
        type="text"
        name="task"
        placeholder="New Todo"
        value={formData.task}
        onChange={handleChange}
      />
      <button>Add Todo</button>
    </form>
  );
};

export default NewTodoForm;
