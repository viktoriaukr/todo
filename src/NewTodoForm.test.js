import React from "react";
import { render } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

test("render", () => {
  render(<NewTodoForm />);
});

test("matches snapshot", () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});
