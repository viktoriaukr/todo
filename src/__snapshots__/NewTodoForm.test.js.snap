// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="task"
    >
      Todo
    </label>
    <input
      id="task"
      name="task"
      placeholder="New Todo"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
